packages/@aws-cdk/aws-synthetics/lib/canary.ts

```diff
@@ -0,0 +1,254 @@
+   * If you provide a Role, you must add the required permissions.
+   *
+   * @see required permissions: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-synthetics-canary.html#cfn-synthetics-canary-executionrolearn
+   *
+   * @default - A unique role will be generated for this canary.
+   * Both supplied and generated roles can always be changed by calling 'addToRolePolicy'.
+   */
+  readonly role?: iam.IRole;
+
+  /**
+   * How long the canary will be in a 'RUNNING' state. For example, if you set `timeToLive` to be 1 hour and `schedule` to be `rate(10 minutes)`,
+   * your canary will run at 10 minute intervals for an hour, for a total of 6 times.
+   *
+   * @default - no limit
+   */
+  readonly timeToLive?: cdk.Duration;
+
+  /**
+   * How often the canary runs. For example, if you set `schedule` to `rate(10 minutes)`, then the canary will run every 10 minutes.
+   *
+   * @default 'rate(5 minutes)'
+   */
+  readonly schedule?: Schedule;
+
+  /**
+   * Whether or not the canary should start after creation. If set to false, then the canary will not start.
+   *
+   * @default true
+   */
+  readonly startAfterCreation?: boolean;
+
+  /**
+   * How many days should successful runs be retained.
+   *
+   * @default - Duration.days(31)
+   */
+  readonly successRetentionPeriod?: cdk.Duration;
+
+  /**
+   * How many days should failed runs be retained.
+   *
+   * @default - Duration.days(31)
+   */
+  readonly failureRetentionPeriod?: cdk.Duration;
+
+  /**
+   * The name of the canary. Be sure to give it a descriptive name that distinguishes it from
+   * other canaries in your account.
+   *
+   * Do not include secrets or proprietary information in your canary name. The canary name
+   * makes up part of the canary ARN, which is included in outbound calls over the internet.
+   * @see https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/servicelens_canaries_security.html
+   *
+   * @default A unique name will be generated
+   */
+  readonly canaryName?: string;
+
+  /**
+   * Specify the endpoint that you want the canary code to hit. Alternatively, you can specify
+   * your own canary script to run.
+   *
+   * 🚧 TODO: implement
+   */
+  //readonly test: Test;
+
+}
+
+export class Canary extends cdk.Resource {
+  /**
+   * Execution role associated with this Canary.
+   */
+  public readonly role: iam.IRole;
+
+  /**
+   * Canary ID
+   * @attribute
+   */
+  public readonly canaryId: string;
+
+  /**
+   * Canary State
+   * @attribute
+   */
+  public readonly canaryState: string;
+
+  /**
+   * Canary Name
+   * @attribute
+   */
+  public readonly canaryName: string;
+
+  /**
+   * Bucket where data from each canary run is stored
+   */
+  private readonly artifactBucket: s3.IBucket;
+
+  constructor(scope: cdk.Construct, id: string, props: CanaryProps) {
+    super(scope, id, {
+      physicalName: props.canaryName || cdk.Lazy.stringValue({
+        produce: () => this.generateUniqueName(),
```

Comment:
I have this as late-bound because I cannot sanitize `this.node.uniqueId` with string manipulation before it is created.

Reply:
>Edit: Having talked to @NetaNir, this is a necessity not because of string manipulation issues but because I cannot call this inside of super as it does not exist yet.

Ohhhh yeah `this.node` is created by `super`. Obviously. 🤦🏻‍♂️